### PR TITLE
fix: #65 - change where statments for gathering passwords

### PR DESCRIPTION
**Pull request by techmaved**
## Motivation

#65 shows a bug that where a password is displayed in the overview but its according to the database in trash and favorite. There was a flaw that should be fixed

## Changes

b06c36b49b1cebde6b685d5eb3cef297913c9630 - moved declaration for results to a switch statement. removed flaw also, hopefully

## Tests done

Created passwords and moved them to trash and favorite and checked. Paste this into your local OmegaPass and test to verify:
```sql
INSERT INTO pwmanager.passwords (user_id, website, username, password, id, trash, trashDate, favorite) VALUES ('8XEiMo3U4tQB', 'trash = 0', 'favorite = 0', 'a', '02a5d51e-f227-11ed-9423-0242ac12', null, null, null);
INSERT INTO pwmanager.passwords (user_id, website, username, password, id, trash, trashDate, favorite) VALUES ('8XEiMo3U4tQB', 'trash = 1', 'favorite = 0', 'a', '1579b091-f55d-11ed-9423-0242ac12', 1, '2023-05-18 18:31:28', null);
INSERT INTO pwmanager.passwords (user_id, website, username, password, id, trash, trashDate, favorite) VALUES ('8XEiMo3U4tQB', 'trash = 0', 'favorite = 1', 'a', '3b5855c0-f358-11ed-9423-0242ac12', null, null, 1);
INSERT INTO pwmanager.passwords (user_id, website, username, password, id, trash, trashDate, favorite) VALUES ('8XEiMo3U4tQB', 'trash = 1', 'favorite = 1', 'a', '4067040d-f40d-11ed-9423-0242ac12', 1, '2023-05-18 18:31:32', 1);

```
Change the user_id of course or change the id of your user

## TODO

- [x] I've assigned this PR to myself
- [x] I've labeled this PR correctly

### CODE REVIEW

**Comment by techmaved** [comment]
@lvlcn-t **Please also test if passwords of other users appears, or change the user_id's of the example ones** Had a bug there also with my implementation

**Comment by lvlcn-t** [comment]
@techmaved Is this still a draft? If not I'd review it since I tested many cases and it worked just fine.

File: db.php [hunk 1]
```diff
@@ -91,25 +91,55 @@ public function get_all_entries($userid, $mode = null) {
...
+                    'username',
+                    'password',
+                    'id'
+                ], [
+                    'user_id' => $userid,
+                    'trash' => true,
+                    'OR' => [
+                        'favorite' => true,
+                        'favorite' => null,
+                        'trash' => true,
```
Comment (lvlcn-t) [inline db.php:107]: Isn't that trash condition redundant because it is already included in the main condition `'trash' => true`?

File: db.php [hunk 2]
```diff
@@ -91,25 +91,55 @@ public function get_all_entries($userid, $mode = null) {
...
+                    'username',
+                    'password',
+                    'id'
+                ], [
+                    'user_id' => $userid,
+                    'trash' => null,
+                    'OR' => [
+                        'favorite' => true,
+                        'favorite' => null,
+                        'trash' => null,
```
Comment (lvlcn-t) [inline db.php:137]: Similar as above

**Review by lvlcn-t — COMMENTED** [review]
few minor things

File: db.php (hunk 1 above)
Comment (techmaved) [reply db.php:107]: You would think so, but when you try to remove it, there are not all passwords displayed.

Comment (lvlcn-t) [reply db.php:107]: Ok weird.
But it as long as the code works we can just keep it in our minds and investigate this issue later.

**Review by lvlcn-t — APPROVED** [review]
We need to keep the code redundancy issue in mind for investigating it.

**Pull request merged**